pilco/controllers.py

```diff
@@ -60,45 +67,58 @@ def randomize(self):
         self.b.assign(mean + sigma*np.random.normal(size=self.b.shape))
 
 
-class FakeGPR(gpflow.Parameterized):
-    def __init__(self, X, Y, kernel):
-        gpflow.Parameterized.__init__(self)
-        self.X = gpflow.Param(X)
-        self.Y = gpflow.Param(Y)
-        self.kern = kernel
+class FakeGPR(gpflow.Module):
+    def __init__(self, data, kernel, X=None):
+        gpflow.Module.__init__(self)
+        if X is None:
+            self.X = Parameter(data[0], name="DataX", dtype=gpflow.default_float())
+        else:
+            self.X = X
+        self.Y = Parameter(data[1], name="DataY", dtype=gpflow.default_float())
+        self.data = [self.X, self.Y]
+        self.kernel = kernel
         self.likelihood = gpflow.likelihoods.Gaussian()
+        self.likelihood.variance.assign(0.0001)
+        set_trainable(self.likelihood.variance, False)
```

Comment:
Did we have numerical problems in the absence of this before?

Either way, I think I will put the variance as a parameter with a default value in `__init__` for more visibility.

Reply:
Yes, there were cases where this collapsed to ~1e-10 and a matrix inversion later on failed.